data_model/src/events/time.rs

```diff
@@ -104,6 +109,29 @@ impl EventFilter {
     }
 }
 
+/// Trigger execution time
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialOrd,
+    Ord,
+    PartialEq,
+    Eq,
+    Encode,
+    Decode,
+    Serialize,
+    Deserialize,
+    IntoSchema,
+    Hash,
+)]
+pub enum ExecutionTime {
```

Comment:
This is an option-like enum. It should use the niche available in the other data type to encode the discriminant, but I would double check.

Reply:
Yep. The sizes are identical even in Debug builds.